src/auth/strategies/oidc.strategy.ts

```diff
@@ -77,12 +83,19 @@ export class OidcStrategy extends PassportStrategy(Strategy, "oidc") {
       accessGroupProperty: oidcConfig?.accessGroupProperty,
       payload: userinfo,
     };
-    userProfile.accessGroups =
+    const accessGroups =
       await this.accessGroupService.getAccessGroups(userPayload);
 
+    if (userProfile.groups && userProfile.groups.length > 0) {
+      userProfile.accessGroups = [...accessGroups, ...userProfile.groups];
+    } else {
+      userProfile.accessGroups = accessGroups;
+    }
+    delete userProfile.groups;
+
     const userFilter: FilterQuery<UserDocument> = {
       $or: [
-        { username: `oidc.${userProfile.username}` },
```

Comment:
this change requires data migrations for existing installations or careful adoption of the username configuration as the username gets prefixed with `oidc.` and you remove that prefix.  -> @PSI

We would probably need to augment this even further as `5.7.  Claim Stability and Uniqueness` in [1] states that only the combination of `iss` and `sub` is guaranteed to be unique, so these fields should be part of the userprofile. 

[1] https://openid.net/specs/openid-connect-core-1_0.html#ClaimStability

Reply:
If you check `migrations/20230309125203-update-user-entries.js` we have migration script to replace the prefixes `oidc.` and `ldap.` to empty string long ago. But have never adapted source code to the migration.
The Uniqueness is a good point, thanks for sharing, maybe we can create another issue to address it? Currently, for OIDC users we are using OR condition with `username` and `email` to get the user which seems like a critical issue needs to be fixed

Reply:
@bpedersen2 how about we keep username configurable, and we provide the configuration for backward compatibility.
Regarding the line that you highlighted above, we should again go the configuration route, as the filtering might change from institution to institution. Also here we can provide the configuration for backward compatibility.

Reply:
Upon reading the material, should we set the default configuration for username as `iss` + `sub` ?
Although through configuration and environmental variable it can be changed